temporalio/bridge/metric.py

```diff
@@ -75,6 +75,46 @@ def record(self, value: int, attrs: MetricAttributes) -> None:
         self._ref.record(value, attrs._ref)
 
 
+class MetricHistogramFloat:
+    """Metric histogram using SDK Core."""
+
+    def __init__(
+        self,
+        meter: MetricMeter,
+        name: str,
+        description: Optional[str],
+        unit: Optional[str],
+    ) -> None:
+        """Initialize histogram."""
+        self._ref = meter._ref.new_histogram_float(name, description, unit)
+
+    def record(self, value: float, attrs: MetricAttributes) -> None:
+        """Record value on histogram."""
+        if value < 0:
+            raise ValueError("Metric value must be non-negative value")
+        self._ref.record(value, attrs._ref)
+
+
+class MetricHistogramDuration:
+    """Metric histogram using SDK Core."""
+
+    def __init__(
+        self,
+        meter: MetricMeter,
+        name: str,
+        description: Optional[str],
+        unit: Optional[str],
+    ) -> None:
+        """Initialize histogram."""
+        self._ref = meter._ref.new_histogram_duration(name, description, unit)
+
+    def record(self, value_ms: int, attrs: MetricAttributes) -> None:
```

Comment:
Why not pass through timedelta here too?

Reply:
It's not the most trivial to create these types of objects on the Rust side and there's a cost to creating them on the Rust side (have to capture GIL, etc). I considered converting back to timedelta closer to the user (and I do it in .NET), but I would have to decide whether to create it on the Rust side or Python side (today each getter of a Python buffered metric value is actually backed by Rust values which is neat). The Rust side is a pain, and the Python side is expensive and unnecessary since the user can do it themselves.

But I have left open this possibility. I have an enum of `MetricBufferDurationFormat` and if we want to add a third `TIMEDELTA` enum we can (like we did in .NET).

Reply:
K, that's reasonable. Being able to record them as timedeltas is the main thing that matters and that works.